File: client/iih_stub/iih.ci_stub/iih.ci.ord/s/ICiSrvRefResultServiceImpl.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using xap.mw.serviceframework;
using iih.ci.ord.srvref.d;

namespace iih.ci.ord.i
{
    public class ICiSrvRefResultServiceImpl : ICiSrvRefResultService
    {

        private readonly ServiceInvocation si;

        private readonly string url_r = XapSvrConfig.BaseUrl + "xap.ci.ord/iih.ci.ord.i.ICiSrvRefResultService";

         /// <summary>
        ///     构造函数
        /// </summary>
        public ICiSrvRefResultServiceImpl()
        {
            si = new ServiceInvocationImpl();
            si.url = url_r;
        }

        /// <summarypublic
        /// 查询医嘱服务
        /// </summary>
        /// <param name="dto">医嘱服务参照入口参数数据</param>
        /// <returns>医嘱服务集合</returns>
        public CiSrvRefResultDTO[] getSrvRefResult(CiSrvRefParamDTO dto) 
        {
            CiSrvRefResultDTO[] srvResultDtos = new CiSrvRefResultDTO[0];
            si.url = url_r;
            try
            {
                srvResultDtos = si.invokeList<CiSrvRefResultDTO>("getSrvRefResult", dto);
                return srvResultDtos;
            }
            catch //(Exception e)
            {
                return srvResultDtos;                 
            }
            
            
        }
    }
}

File: server/iih.ci/iih.ci.ord/src/main/java/iih/ci/ord/s/ICiSrvRefResultServiceImpl.java
package iih.ci.ord.s;

import iih.ci.ord.i.ICiSrvRefResultService;
import iih.ci.ord.s.bp.srvref.GetSrvRefRstDatumBP;
import iih.ci.ord.srvref.d.CiSrvRefParamDTO;
import iih.ci.ord.srvref.d.CiSrvRefResultDTO;
import xap.mw.core.annotation.Service;
import xap.mw.core.data.BizException;
import xap.mw.core.service.constant.Binding;

/**
 * 医嘱服务实现类
 * @author hums
 *
 */
@Service(serviceInterfaces={ICiSrvRefResultService.class}, binding=Binding.JSONRPC)
public class ICiSrvRefResultServiceImpl implements ICiSrvRefResultService {

	@Override
	public CiSrvRefResultDTO[] getSrvRefResult(CiSrvRefParamDTO dto) throws BizException {
		
		GetSrvRefRstDatumBP bp = new GetSrvRefRstDatumBP();
		CiSrvRefResultDTO[] resultDto = bp.exec(dto);
		return resultDto;
	}

}
